submissions/ses3332002/dom/script.js

```diff
@@ -0,0 +1,260 @@
+      ],
+      "bgColor": 0
+    },
+    {
+      "title": "Dangerous (1991)",
+      "cover": "dangerous",
+      "songs": [
+        "Jam",
+        "Why You Wanna Trip on Me",
+        "In the Closet",
+        "She Drives Me Wild",
+        "Remember the Time",
+        "Can't Let Her Get Away",
+        "Heal the World",
+        "Black or White",
+        "Who Is It",
+        "Give in to Me",
+        "Will You Be There",
+        "Keep the Faith",
+        "Gone Too Soon",
+        "Dangerous"
+      ],
+      "bgColor": 29
+    },
+    {
+      "title": "HIStory: Past, Present and Future, Book I (1995)",
+      "cover": "history",
+      "songs": [
+        "Scream",
+        "They Don’t Care About Us",
+        "Stranger in Moscow",
+        "This Time Around",
+        "Earth Song",
+        "D.S.",
+        "Money",
+        "Come Together",
+        "You Are Not Alone",
+        "Childhood",
+        "Tabloid Junkie",
+        "2 Bad",
+        "HIStory",
+        "Little Susie",
+        "Smile"
+      ],
+      "bgColor": 332
+    },
+    {
+      "title": "Invincible (2001)",
+      "cover": "invincible",
+      "songs": [
+        "Unbreakable",
+        "Heartbreaker",
+        "Invincible",
+        "Break of Dawn",
+        "Heaven Can Wait",
+        "You Rock My World",
+        "Butterflies",
+        "Speechless",
+        "2000 Watts",
+        "You Are My Life",
+        "Privacy",
+        "Don't Walk Away",
+        "Cry",
+        "The Lost Children",
+        "Whatever Happens",
+        "Threatened"
+      ],
+      "bgColor": 215
+    },
+    {
+      "title": "Michael (2010)",
+      "cover": "michael",
+      "songs": [
+        "Hold My Hand",
+        "Hollywood Tonight",
+        "Keep Your Head Up",
+        "(I Like) The Way You Love Me",
+        "Monster",
+        "Best of Joy",
+        "Breaking News",
+        "(I Can't Make It) Another Day",
+        "Behind the Mask",
+        "Much Too Soon"
+      ],
+      "bgColor": 8
+    },
+    {
+      "title": "Xscape (2014)",
+      "cover": "xscape",
+      "songs": [
+        "Love Never Felt So Good",
+        "Chicago",
+        "Loving You",
+        "A Place with No Name",
+        "Slave to the Rhythm",
+        "Do You Know Where Your Children Are",
+        "Blue Gangsta",
+        "Xscape"
+      ],
+      "bgColor": 33
+    }
+  ];
+
+  // definition of page elements
+  var doc = document.querySelector("body");
+  var navTab = document.querySelector(".navigation");
+  var infoTab = document.querySelector(".info");
+
+  // filling the menu block
+  navTab.innerHTML = "";
+  var navTabBtn;
+  for (var i = 0; i < alboms.length; i++) {
+    navTabBtn = document.createElement("button");
+    navTabBtn.classList.add("navigation_btn");
+    navTabBtn.innerText = alboms[i].title;
+    navTab.append(navTabBtn);
+  };
+  var navTabBtns = document.querySelectorAll(".navigation_btn");
+  var navTabBtnsArray = Array.prototype.slice.call(navTabBtns);
```

Comment:
`Array.prototype.slice.call` - no need for such dirty things.

You can use [`Array.from`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from).

And I don't see any reason of why you make copy. So, why do you need shallow copy of these elements?

Reply:
I need to define which button is clicked while I listen to "click" event on parent block, so I found "Array.prototype.slice.call" to convert HTML-collection into array and use array's method "indexOf".
I'll remake it with Array.from.
I can make this code shorter, but I afraid it will be less "readable" for review. Is it OK?

Reply:
`Array.from` will be more readable then `Array.prototype.slice.call`.

Also, you should avoid this `prototype.{method}.call` things in the future. Most such hacks are outdated and you can use other methods or build your code using another logic in which such things will be not needed.

Reply:
Well, it was the first answer that Google gave to me, and it works :)

Reply:
ready